src/doc/tarpl/lifetimes.md

```diff
@@ -0,0 +1,216 @@
+% Lifetimes
+
+Rust enforces these rules through *lifetimes*. Lifetimes are effectively
+just names for scopes somewhere in the program. Each reference,
+and anything that contains a reference, is tagged with a lifetime specifying
+the scope it's valid for.
+
+Within a function body, Rust generally doesn't let you explicitly name the
+lifetimes involved. This is because it's generally not really *necessary*
+to talk about lifetimes in a local context; rust has all the information and
+can work out everything. It's also a good thing because the scope of a borrow
+is often significantly smaller than the scope its referent is *actually* valid
```

Comment:
nit: trailing preposition here. Consider rephrasing to simplify the reader's experience.